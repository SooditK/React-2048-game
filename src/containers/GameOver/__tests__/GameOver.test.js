import React from 'react';
import renderer from 'react-test-renderer';
import Enzyme, { mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { GameOver } from '..';

Enzyme.configure({ adapter: new Adapter() });

describe('<GameOver />', () => {
  it('component render', () => {
    const fn = () => {};
    let gameover = renderer
      .create(<GameOver score={123} gameOver={false} onReset={fn} />)
      .toJSON();
    expect(gameover).toMatchSnapshot();
    gameover = renderer
      .create(<GameOver score={9999} gameOver onReset={fn} />)
      .toJSON();
    expect(gameover).toMatchSnapshot();
  });

  it('mount', () => {
    const fn = jest.fn();
    const gameover = mount(
      <GameOver score={123} gameOver={false} onReset={fn} />
    );
    gameover.find("[alt='reset']").simulate('click');
    expect(fn.mock.calls.length).toBe(1);
  });
});
