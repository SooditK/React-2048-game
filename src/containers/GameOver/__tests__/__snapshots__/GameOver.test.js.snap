// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GameOver /> component render 1`] = `
<div>
  <div
    className="hidden"
  >
    <div
      className="overlay"
    />
    <div
      className="modal center"
    >
      <div>
        <div
          className="gameover"
        >
          <div
            className="text"
          >
             
            SCORE
             
          </div>
          <div
            className="banner"
          >
            <p>
              123
            </p>
          </div>
        </div>
        <div
          className="buttonWrapper"
        >
          <button
            className="button"
            onClick={[Function]}
          >
            <img
              alt="reset"
              src="reset.svg"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<GameOver /> component render 2`] = `
<div>
  <div
    className=""
  >
    <div
      className="overlay"
    />
    <div
      className="modal center"
    >
      <div
        className="firework"
      >
        <div
          className="before"
        />
        <div
          className="after"
        />
      </div>
      <div>
        <div
          className="gameover"
        >
          <div
            className="text"
          >
             
            SCORE
             
          </div>
          <div
            className="banner"
          >
            <p>
              9999
            </p>
          </div>
        </div>
        <div
          className="buttonWrapper"
        >
          <button
            className="button"
            onClick={[Function]}
          >
            <img
              alt="reset"
              src="reset.svg"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
